lib/voxpupuli/release/rake_tasks.rb

```diff
@@ -52,3 +52,116 @@
     fail "Unable to find a CHANGELOG.md entry for the #{v} release."
   end
 end
+
+desc "Prepare a release"
+task "release:prepare" do
+  v = Blacksmith::Modulefile.new.version
+  Rake::Task["release:porcelain:changelog"].invoke
+  Rake::Task["strings:generate:reference"].invoke if File.exist?('REFERENCE.md')
+  puts <<~MESSAGE
+
+    Please review these changes and commit them to a new branch:
+
+        git checkout -b release-#{v}
+        git commit -m "Release #{v}"
+
+    Then open a Pull-Request and wait for it to be reviewed and merged).
+  MESSAGE
+end
+
+begin
+  require 'github_changelog_generator/task'
+  require 'puppet_blacksmith'
+rescue LoadError
+  desc "Dummy"
+  task "release:porcelain:changelog" do
+    puts "Skipping CHANGELOG.md generation.  Ensure github_changelog_generator is present if you expected it to be generated."
+  end
+else
+  class GCGConfig
+    def self.user=(user)
+      @user = user
+    end
+
+    def self.user
+      @user || project.split(%r{[-/]}).first
+    end
+
+    def self.project=(project)
+      @project = project
+    end
+
+    def self.project
+      @project || metadata['name']
+    end
+
+    def self.metadata
+      @metadata ||= Blacksmith::Modulefile.new.metadata
+    end
+
+    def self.tag_pattern=(tag_pattern)
+      @tag_pattern = tag_pattern
+    end
+
+    def self.tag_pattern
+      @tag_pattern || 'v%s'
+    end
+
+    def self.future_release
+      format(tag_pattern, metadata['version'])
+    end
+  end
+
+  task "release:porcelain:changelog" do
+    # This is taken from lib/github_changelog_generator/task
+    # The generator cannot be used because we want to lazyly evaluate
+    # GCGConfig.user which might be overrider in the module Rakefile.
+    options = GitHubChangelogGenerator::Parser.default_options
+    options[:user] = GCGConfig.user
+    options[:project] = GCGConfig.project
+    options[:future_release] = GCGConfig.future_release
+    options[:header] = <<~HEADER.chomp
+      # Changelog
+
+      All notable changes to this project will be documented in this file.
+      Each new release typically also includes the latest modulesync defaults.
+      These should not affect the functionality of the module.
+    HEADER
+    options[:exclude_labels] = %w{duplicate question invalid wontfix wont-fix modulesync skip-changelog}
+    generator = GitHubChangelogGenerator::Generator.new(options)
+    log = generator.compound_changelog
+    output_filename = options[:output].to_s
+    File.write(output_filename, log)
+
+    # Workaround for https://github.com/github-changelog-generator/github-changelog-generator/issues/715
+    require 'rbconfig'
+    if RbConfig::CONFIG['host_os'].match?(/linux/)
+      puts 'Fixing line endings...'
+      changelog_file = 'CHANGELOG.md'
```

Comment:
Isn't this the earlier mentioned `output_filename`? And if so, can we avoid a write + read + write and only write once?